docs/guides/stateful_modules.rst

```diff
@@ -0,0 +1,28 @@
+Using statful modules
```

Comment:
I wonder whether we should call it "Stateful Modules" or "Bound Modules". I think Stateful Modules are a bit confusing since Flax Modules are generally stateless (and we say this *a lot*).

Also, as a general comment, I wonder whether we need to add a verb before every guide, or whether it would be clearer to just use the noun phrase. So "Using Bound Modules" --> "Bound Modules", "Applying batch normalization" --> "Batch Normalization", etc. WDYT?

```suggestion
Using stateful modules
```